common/src/main/java/org/opensearch/ml/common/conversational/ActionConstants.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.common.conversational;
+
+/**
+ * Constants for conversational actions
+ */
+public class ActionConstants {
+
+    /** name of conversation Id field in all responses */
+    public final static String CONVERSATION_ID_FIELD = "conversation_id";
+
+    /** name of list of conversations in all responses */
+    public final static String RESPONSE_CONVERSATION_LIST_FIELD = "conversations";
+    /** name of list on interactions in all responses */
+    public final static String RESPONSE_INTERACTION_LIST_FIELD = "interactions";
+    /** name of interaction Id field in all responses */
+    public final static String RESPONSE_INTERACTION_ID_FIELD = "interaction_id";
+
+    /** name of conversation name in all requests */
+    public final static String REQUEST_CONVERSATION_NAME_FIELD = "name";
+    /** name of maxResults field name in all requests */
+    public final static String REQUEST_MAX_RESULTS_FIELD = "max_results";
+    /** name of nextToken field name in all messages */
+    public final static String NEXT_TOKEN_FIELD = "next_token";
+    /** name of input field in all requests */
+    public final static String INPUT_FIELD = "input";
+    /** name of prompt field in all requests */
+    public final static String PROMPT_FIELD = "prompt";
+    /** name of AI response field in all respopnses */
+    public final static String AI_RESPONSE_FIELD = "response";
+    /** name of agent field in all requests */
+    public final static String AI_AGENT_FIELD = "agent";
```

Comment:
Are you going to build a similar Agent like #1161 ?

Reply:
Not exactly - we're considering applications that could use multiple agents, of which perhaps our RAG stuff might be one, or an 1161 Agent might be one. But we want to be general enough to be able to support all kinds of apps, so that someone who never wanted to use OS for the AI part but still wants a persistent conversational memory storage solution might pick it up. Also, if this can sit underneath the 1161-style agents, that's a plus, right?

Side note, but I imagine CoT agents using this by spawning a conversation per chain, where the 'routing agent' acts as the 'end-user' and its tools as the various AI Agents that it's hitting. Then the 'real' interactions include just the final input and output, with interaction metadata holding sub-conversation id. hmm, gets a little hairy when you want to delete a conversation in that model. Interesting

Reply:
> Also, if this can sit underneath the 1161-style agents, that's a plus, right?

yes, I think this is also what I mentioned before on last community meeting that conversation history could be one type of memory.